docs/rfc/0003-importer-interface.md

```diff
@@ -0,0 +1,62 @@
+# Importer Interface RFC
+
+- Contribution Name: Importer Interface/Plugin Framework
+- Implementation Owner: Ganesh Vanahalli
+
+## Problem Statement
+
+Users would like to choose how to fetch blocks, either by downloading from an algod rest endpoint or directly from the gossip network or read blocks from a file which was previously written by the exporter.
+
+The current implementation of indexer only allows downloading blocks sequentially from an algod rest endpoint starting at the round number of recently written block to postgres db and users cannot choose a specific block to download.
+
+## Proposal
+
+Importer interface allows users to fetch any particular block either from algod rest endpoint (or directly from the network or from a file written to by the exporter plugin). Current interface also stores round number of the latest fetched block. 
+
+### Plugin Interface
+A sample interface is shown below:
+
+```GO
+package importerplugin
+
+// Importer defines the interface for importer plugin
+type Importer interface {
+	// Algod returns the importer's algod client
+	Algod() *algod.Client
+
+	// Round returns the recent downloaded block
+	Round() uint64
```

Comment:
I actually don't know if an `Importer` should be keeping track of which rounds it has imported in the past. It seems like one or both of a `Processor` plugin or `Exporter` which require ordered processing of blocks would be the ones maintaining that state. Then whatever framework is coordinating local ledger catchup, postgres (or other exporter) writes, etc. would just request whatever blocks it needs from the Importer without the Importer caring which rounds it fetched previously.

Reply:
Makes sense, the importer doesn't necessarily need to store any history